sdk/storage/azure-storage-stress/src/main/java/com/azure/storage/stress/TelemetryHelper.java

```diff
@@ -195,9 +196,22 @@ private void trackCancellation(Instant start, Span span) {
     private void trackFailure(Instant start, Throwable e, Span span) {
         Throwable unwrapped = Exceptions.unwrap(e);
 
+        // Check if the message contains "NativeIoException" and the unwrapped exception is not already a NativeIoException
+
+        // Check if the unwrapped exception is a RuntimeException
+        if (unwrapped instanceof RuntimeException) {
+            String message = unwrapped.getMessage();
+            if (message.contains("NativeIoException")) {
+                unwrapped = new io.netty.channel.unix.Errors.NativeIoException("recvAddress", Errors.ERRNO_ECONNRESET_NEGATIVE);
+            } else if (message.contains("TimeoutException")) {
+                unwrapped = new TimeoutException(message);
+            }
+            // may need to add more known exceptions here
+        }
+
         span.recordException(unwrapped);
         span.setAttribute(ERROR_TYPE_ATTRIBUTE, unwrapped.getClass().getName());
-        span.setStatus(StatusCode.ERROR, unwrapped.getMessage());
+        span.setStatus(StatusCode.ERROR, unwrapped.getMessage()); // add check here to see if its a connection reset exception/NativeIoException
```

Comment:
We should be able to remove this comment as I'm not sure if it applies to what the line does.